src/main/java/com/fittracker/fittracker/exception/RestExceptionHandler.java

```diff
@@ -30,6 +30,11 @@ ResponseEntity<ErrorResponse> handleException(WeightNotFoundException e) {
         return new ResponseEntity<>(errorResponseMapper.map(e), NOT_FOUND);
     }
 
+    @ExceptionHandler(ProductNotFoundException.class)
+    ResponseEntity<ErrorResponse> handleException(ProductNotFoundException e) {
```

Comment:
If we had more entites we could create parent `NotFoundException` for handling all 404s. But for now let's leave it as it is.

Reply:
Lets stick to the "Rule of Three":
```
It states that you are allowed to copy and paste the code once, but that when the same code is replicated three times, it should be extracted into a new procedure. The rule was introduced by Martin Fowler in his text "Refactoring" and attributed to Don Roberts.
```

Reply:
yes, that was my thinking exactly :)